### PR TITLE
Fix composure effect accuracy bonus calculation

### LINKED ISSUE #6330: RDM JA Composure Accuracy Buff Value Incorrect
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my issue will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.

## OS / platform the server is running (if known)

Windows Server

## Branch affected by issue

`base`

## Steps to reproduce

Change job to RDM50 or greater
Use !getstats on yourself and note accuracy boost
Use job ability composure
Use !getstats on yourself and note accuracy boost

## Expected behavior

The accuracy boost should be 25~50 depending on level, plus up to an additional 20 from job point bonuses.  It is currently a flat 15 plus job point bonuses.

### PR DESCRIPTION
**Pull request by lefneer311**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Updates the scripts/effects/composure.lua to be able to cast the method getMainLevel() for calculating appropriate accuracy bonus, then applies and removes when appropriate.

Closes #6330 

## Steps to test these changes

Change job to RDM50 or greater
Use !getstats on yourself and note accuracy boost
Use job ability composure
Use !getstats on yourself and note accuracy boost

Your bonus accuracy should be roughly 1/2 of RDM Main level (35 at 70, 45 at 90, etc.)

### CODE REVIEW

File: scripts/effects/composure.lua [hunk 1]
```diff
@@ -7,18 +7,22 @@
...
 effectObject.onEffectGain = function(target, effect)
-    local jpValue = target:getJobPointLevel(xi.jp.COMPOSURE_EFFECT)
+    local player = target:getMaster() or target
+    local mLevel = player:getMainLvl()
+    local jpValue = player:getJobPointLevel(xi.jp.COMPOSURE_EFFECT)
+    local accPower = math.floor(((24 * mLevel) + 74) / 49)
+    effect:setPower(accPower + jpValue)
 
-    target:addMod(xi.mod.ACC, 15 + jpValue)
+    player:addMod(xi.mod.ACC, effect:getPower())
```
Comment (Xaver-DaRed) [inline scripts/effects/composure.lua]:
```lua
local power = math.floor((24 * target:getMainLvl() + 74) / 49) + target:getJobPointLevel(xi.jp.COMPOSURE_EFFECT)

effect:addMod(xi.mod.ACC, power)
```

Comment (Xaver-DaRed) [reply scripts/effects/composure.lua]: Thats all you need

File: scripts/effects/composure.lua [hunk 2]
```diff
@@ -7,18 +7,22 @@
...
 
 effectObject.onEffectTick = function(target, effect)
 end
 
 effectObject.onEffectLose = function(target, effect)
-    local jpValue = target:getJobPointLevel(xi.jp.COMPOSURE_EFFECT)
+    local player = target:getMaster() or target
 
-    target:delMod(xi.mod.ACC, 15 + jpValue)
+    player:delMod(xi.mod.ACC, effect:getPower())
```
Comment (Xaver-DaRed) [inline scripts/effects/composure.lua]: And then this 2 lines are unnecesary

File: scripts/effects/composure.lua (hunk 1 above)
Comment (lefneer311) [reply scripts/effects/composure.lua]: Thank you for the review!  I agree.  This approach is a leftover from a more complex implementation that adjusted the bonus accuracy dynamically using onTick, which I realized didn't reflect retail.  I'll clean it up as you've recommended here.

**Review by Xaver-DaRed — DISMISSED** [review]

**Comment by Xaver-DaRed** [comment]
Please be sure to read the style guide and apply it to the PR

**Pull request closed without merging**